File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/core/usecase/commons/SystemResults.java
package com.mycompany.myshop.testkit.dsl.core.usecase.commons;

import com.mycompany.myshop.testkit.driver.port.dtos.error.SystemError;
import com.mycompany.myshop.testkit.common.Result;

public final class SystemResults {
    private SystemResults() {
        // Utility class
    }

    public static <T> Result<T, SystemError> success(T value) {
        return Result.success(value);
    }

    public static Result<Void, SystemError> success() {
        return Result.success();
    }

    public static <T> Result<T, SystemError> failure(String message) {
        return Result.failure(SystemError.of(message));
    }

    public static <T> Result<T, SystemError> failure(SystemError error) {
        return Result.failure(error);
    }
}

File: system-test/dotnet/Dsl.Core/UseCase/Commons/SystemResults.cs
using Common;
using Driver.Port.Dtos.Error;

namespace Dsl.Core.UseCase.Commons;

public static class SystemResults
{
    public static Result<T, SystemError> Success<T>(T value)
    {
        return Result<T, SystemError>.Success(value);
    }

    public static Result<VoidValue, SystemError> Success()
    {
        return Result.Success<SystemError>();
    }

    public static Result<T, SystemError> Failure<T>(string message)
    {
        return Result<T, SystemError>.Failure(SystemError.Of(message));
    }

    public static Result<T, SystemError> Failure<T>(SystemError error)
    {
        return Result<T, SystemError>.Failure(error);
    }
}
